cli/src/commands/extensions/mod.rs

```diff
@@ -273,7 +274,24 @@ fn ask_permissions(extension: &Extension) -> Result<()> {
         }
     }
 
-    print_permissions_list("Read", "path", permissions.read.get());
+    // Calculate effective read permissions
+    let read = match (&permissions.read, &permissions.run) {
+        (Permission::Boolean(true), _) => Permission::Boolean(true),
+        (_, Permission::Boolean(true)) => Permission::Boolean(true),
+        (Permission::Boolean(false), run) => run.clone(),
+        (read, Permission::Boolean(false)) => read.clone(),
+        (Permission::List(ref read), Permission::List(ref run)) => {
```

Comment:
You don't need `ref` here, these are already borrowed due to the `&permissions.read`.